.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,66 @@
+name: Java CI
+
+on: [pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: 'postgres'
+          POSTGRES_DB: test_db
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+    - name: Create test MySQL database
+      id: create_test_db
+      uses: mirromutth/mysql-action@v1.1
+      with:
+        host port: 3306
+        container port: 3306
+        mysql database: 'test_db'
+        mysql root password: 'root'
```

Comment:
I think we can probably run MySQL as a service too, along the lines of https://firefart.at/post/using-mysql-service-with-github-actions/, including the "wait for availability" check